### PR TITLE
Add apollo-codegen 0.16

### PR DESCRIPTION
**Pull request by bradleyayers**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/apollo-codegen/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Type definitions for apollo-codegen 0.16
+// Project: https://github.com/apollographql/apollo-codegen
+// Definitions by: Bradley Ayers <https://github.com/bradleyayers>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
```
Comment (plantain-00) [inline types/apollo-codegen/index.d.ts]: It can be Typescript 2.5 here

File: types/apollo-codegen/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// Type definitions for apollo-codegen 0.16
+// Project: https://github.com/apollographql/apollo-codegen
```
Comment (plantain-00) [inline types/apollo-codegen/index.d.ts]: This project doesn't bundle types, but it is written by Typescript, just adding `"declaration": true` in `tsconfig.json` and `types` in `package.json` can be a simpler way, so you can try to ask the package maintainer whether it is doable.

Comment (bradleyayers) [reply types/apollo-codegen/index.d.ts]: The project is only partially converted to TypeScript (there's lots of `any` types). It's on their roadmap to get to the point where they can publish declarations, but they're not there yet.

File: types/apollo-codegen/index.d.ts (hunk 1 above)
Comment (bradleyayers) [reply types/apollo-codegen/index.d.ts]: I thought it would be desirable to have the oldest compatible version?

**Review by plantain-00 — APPROVED** [review]

**Pull request merged**